AsyncDisplayKit/ASMapNode.mm

```diff
@@ -159,6 +161,15 @@ - (void)setRegion:(MKCoordinateRegion)region
 
 - (void)takeSnapshot
 {
+  // do not trigger snapshot before layout
+  ASLayout *layout = self.calculatedLayout;
+  if (layout == nil || CGSizeEqualToSize(CGSizeZero, layout.size)) {
```

Comment:
Once the layout is set, whether or not it changes, this condition should always proceed into the snapshot taking process, therefore it shouldn't have any impact except for when the layout is nil or zero sized.